server/src/main/java/com/cloud/storage/snapshot/SnapshotManagerImpl.java

```diff
@@ -381,6 +381,24 @@ public Snapshot createSnapshot(Long volumeId, Long policyId, Long snapshotId, Ac
         return snapshot;
     }
 
+    @Override
+    public Snapshot archiveSnapshot(Long snapshotId) {
+        SnapshotInfo snapshotOnPrimary = snapshotFactory.getSnapshot(snapshotId, DataStoreRole.Primary);
+
+        if (snapshotOnPrimary == null || !snapshotOnPrimary.getStatus().equals(ObjectInDataStoreStateMachine.State.Ready)) {
+            throw new CloudRuntimeException("Can only archive snapshots present on primary storage. " +
+                    "Cannot find snapshot " + snapshotId + " on primary storage");
+        }
+
+        SnapshotInfo snapshotOnSecondary = snapshotSrv.backupSnapshot(snapshotOnPrimary);
+        snapshotSrv.deleteSnapshot(snapshotOnPrimary);
```

Comment:
@syed, I think it would be better to check the status of deleting the snapshot on the primary storage. Due to any reason, if it fails to delete then the user will be informed with a proper error message. Thanks.